lib/pbench/test/unit/server/auth/test_auth.py

```diff
@@ -712,6 +714,37 @@ def test_verify_auth_oidc_user_admin(
         assert user.roles == ["ROLE", "ADMIN"]
         assert user.username == "friend"
 
+    def test_verify_auth_oidc_user_duplicate(
+        self, monkeypatch, server_config, db_session, rsa_keys, make_logger
+    ):
+        """Verify behavior when an OIDC username already exists with a
+        different UUID."""
+        client_id = "us"
+        token, expected_payload = gen_rsa_token(client_id, rsa_keys["private_key"])
+
+        # Mock the Connection object and generate an OpenIDClient object,
+        # installing it as Auth module's OIDC client.
+        config = mock_connection(
+            monkeypatch, client_id, public_key=rsa_keys["public_key"]
+        )
+        oidc_client = OpenIDClient.construct_oidc_client(config)
+        monkeypatch.setattr(Auth, "oidc_client", oidc_client)
+
+        app = Flask("test-verify-auth-oidc-user-dup")
+        app.logger = make_logger
+        app.server_config = server_config
+
+        assert not User.query(id="12345")
+        User(username="dummy", id="abcde").add()
+        orig = User.query(id="12345")
+        copy = User.query(id="abcde")
+        assert orig != copy, f"{orig} == {copy}"
```

Comment:
I'm missing something here: doesn't line 737 confirm that there is no user with ID `"12345"`, in which case at line 739 `orig` is going to be set to `None` or `[]` or `""` something false-y?  I mean, sure, `orig` won't match `copy`, but that's not proving much, is it?

Reply:
This made sense at the time. But I suspect what I originally intended was to be sure that the (failed) authorization hadn't changed anything, in which case this is too early.